packages/markedparser-extension/src/index.ts

```diff
@@ -15,6 +15,28 @@ import { IEditorLanguageRegistry } from '@jupyterlab/codemirror';
 import { IMarkdownParser } from '@jupyterlab/rendermime';
 import { marked } from 'marked';
 
+/**
+ * Create a markdown parser
+ *
+ * @param languages Editor languages
+ * @returns Markdown parser
+ */
+export function createMarkdownParser(languages: IEditorLanguageRegistry) {
```

Comment:
This creates an interesting dilemma over on #14102 where I had added the optional `IMermaidMarkdown` token. As this interchange was previously private, that PR would now need to either change this API or do some dirty tricks to hide some of the up-front loading such that it can't be extended.